src/pudl/constants.py

```diff
@@ -1365,23 +518,8 @@
     to keep for those tables (values).
 """
 
-epacems_tables = ("hourly_emissions_epacems")
-"""tuple: A tuple containing tables of EPA CEMS data to pull into PUDL.
-"""
-
-data_sources = (
-    'eia860',
-    'eia861',
-    'eia923',
-    'epacems',
-    'ferc1',
-    'ferc714',
-    # 'pudl'
-)
-"""tuple: A tuple containing the data sources we are able to pull into PUDL."""
-
 # All the years for which we ought to be able to download these data sources
-data_years = {
+data_years: Dict[str, Tuple] = {
```

Comment:
The reason all of these are using Tuples is so that they're immutable and can't get overwritten by functions that are taking them as default arguments. Would it be better to make the dictionaries themselves `ImmutableDict`?  I think most of this information should be managed in some kind of settings class that can also be used for validation too.

Reply:
Yeah, I guess you could build all of the metadata out of immutable classes: int, float, str, None, tuple, frozenset, ImmutableDict. StaticFrame for Data frames? Maybe not in this PR, though...

I am perplexed by the existence of DATA_YEARS. Only 'ferc1' is accessed in the code, and based on the docstring, I would expect it describes a result, not a software behavior.

Reply:
IIRC the switch to tuples was precipitated by a bug in which one of these values was accidentally changed in a transform process, and had unexpected impacts on other parts of the pipeline.

It's totally possible that data years is vestigial, and left over from before we had the current original data management system, where the available data partitions (years, states, etc.) are stored in the raw input datapackage metadata.